lxd/api_cluster.go

```diff
@@ -753,30 +753,37 @@ func clusterPutJoin(d *Daemon, r *http.Request, req api.ClusterPut) response.Res
 			return err
 		}
 
-		// Connect to MAAS
-		url, key := s.GlobalConfig.MAASController()
-		machine := nodeConfig.MAASMachine()
-		err = d.setupMAASController(url, key, machine)
+		d.globalConfigMu.Lock()
+		d.localConfig = nodeConfig
+		d.globalConfigMu.Unlock()
+
+		// Get the current (updated) config.
+		var existingClusterConfig *clusterConfig.Config
+		err = d.db.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+			existingClusterConfig, err = clusterConfig.Load(ctx, tx)
```

Comment:
is existingClusterConfig represent the updated cluster config that needs to be applied to this joining member?

Reply:
Yes. I wasn't sure what to call it since it is the config that is existing in the cluster DB, there shouldn't be any cluster config already on the joining node since it wasn't part of the cluster until the previous few lines.

Reply:
I think call it "currentClusterConfig" or "activeClusterConfig" as "existing" suggested to me it was what existed on this member, what do you think?

Reply:
`currentClusterConfig` sounds good to me :)

Reply:
Done.